facilitator/src/gcp_oauth.rs

```diff
@@ -260,16 +262,19 @@ impl OauthTokenProvider {
         // https://cloud.google.com/iam/docs/reference/credentials/rest/v1/projects.serviceAccounts/generateAccessToken
         let request_url = format!("https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/{}:generateAccessToken",
             service_account_to_impersonate);
-        let auth = format!("Bearer {}", self.ensure_default_account_token()?);
-        let http_response = ureq::post(&request_url)
-            // By default, ureq will wait forever to connect or read.
-            .timeout_connect(10_000) // ten seconds
-            .timeout_read(10_000) // ten seconds
-            .set("Authorization", &auth)
-            .set("Content-Type", "application/json")
-            .send_json(ureq::json!({
+        let request = ureq::post(&request_url)
+            .set(
+                "Authorization",
+                &format!("Bearer {}", self.ensure_default_account_token()?),
+            )
```

Comment:
Can this be passed as `token_provider: self`?

Reply:
No, because here we want to call `ensure_default_account_token()` and not `ensure_oauth_token()` as most callers of `send_json_request()`. In fact passing in `token_provider: self` would cause an infinite recursion! Since this is the only callsite that wants this behaviour, I think it's easier to add the special logic here instead of making `send_json_request()` more complicated.